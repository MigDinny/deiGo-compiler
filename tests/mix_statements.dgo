package main;

var a, b, c int;

func main(){

    for {
        if (a == b || c){
            fmt.Println(abc(a && b, c));
        } else{
            fmt.Println(abc(a, b));
        }
    }

    {
        fmt.Println("Print sem Block");
    };

    {
        fmt.Println("Print com Block");
        fmt.Println("Tem mais de 2 statements");
    };

    {
        fmt.Println("Statements vazios nao contam");
        {};
        {};
    };

    if 1 {};

    if 2 {
        arg,_ = strconv.Atoi(os.Args[3]);
    } else {
        a = 1.2
    }

}

func abc(x int, z float32){
    {
        return x * z % z;
    };
}