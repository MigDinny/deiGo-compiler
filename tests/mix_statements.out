Program
..VarDecl
....Int
....Id(a)
..VarDecl
....Int
....Id(b)
..VarDecl
....Int
....Id(c)
..FuncDecl
....FuncHeader
......Id(main)
......FuncParams
....FuncBody
......For
........Block
..........If
............Or
..............Eq
................Id(a)
................Id(b)
..............Id(c)
............Block
..............Print
................Call
..................Id(abc)
..................And
....................Id(a)
....................Id(b)
..................Id(c)
............Block
..............Print
................Call
..................Id(abc)
..................Id(a)
..................Id(b)
......Print
........StrLit("Print sem Block")
......Block
........Print
..........StrLit("Print com Block")
........Print
..........StrLit("Tem mais de 2 statements")
......Print
........StrLit("Statements vazios nao contam")
......If
........IntLit(1)
........Block
........Block
......If
........IntLit(2)
........Block
..........ParseArgs
............Id(arg)
............IntLit(3)
........Block
..........Assign
............Id(a)
............RealLit(1.2)
..FuncDecl
....FuncHeader
......Id(abc)
......FuncParams
........ParamDecl
..........Int
..........Id(x)
........ParamDecl
..........Float32
..........Id(z)
....FuncBody
......Return
........Mod
..........Mul
............Id(x)
............Id(z)
..........Id(z)